swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/Subscription.java

```diff
@@ -43,21 +45,69 @@ public class Subscription {
   @NotNull @NotEmpty private String id; // required
 
   /**
-   * Enables capability to inherit billing model information from their parent subscription Unused
-   * prior to https://issues.redhat.com/browse/BIZ-629
+   * Enables capability to inherit billing model information from their parent subscription. Unused
+   * prior to <a href="https://issues.redhat.com/browse/BIZ-629">BIZ-629</a>
    */
   private String parentSubscription;
 
   /**
    * defines an "in-the-box" subscription. Considered included from both usage and capacity
    * perspectives.
    */
-  private List<String> includedSubscriptions;
+  private List<String> includedSubscriptions = new ArrayList<>();
 
   private Fingerprint fingerprint;
-  private List<Variant> variants;
+  private List<Variant> variants = new ArrayList<>();
   private BillingWindow billingWindow;
   private String serviceType;
-  private List<Metric> metrics;
+  private List<Metric> metrics = new ArrayList<>();
   private Defaults defaults;
+
+  public static Optional<Subscription> findByServiceType(String serviceType) {
+    return SubscriptionConfigRegistry.getInstance().lookupByServiceType(serviceType);
+  }
+
+  public static Optional<Subscription> findByArch(String arch) {
+    return SubscriptionConfigRegistry.getInstance().lookupByFingerprintArch(arch);
+  }
+
+  public List<String> getMetricIds() {
+    return this.getMetrics().stream()
+        .map(Metric::getId)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  public Optional<Metric> getMetric(String metricId) {
+    return this.getMetrics().stream()
+        .filter(Objects::nonNull)
+        .filter(x -> Objects.equals(x.getId(), metricId))
+        .findFirst();
+  }
+
+  public static Optional<Subscription> findById(String id) {
+    return SubscriptionConfigRegistry.getInstance().getSubscriptions().stream()
+        .filter(subscription -> Objects.equals(subscription.getId(), id))
+        .findFirst();
+  }
+
+  public static List<String> getAllServiceTypes() {
+    return SubscriptionConfigRegistry.getInstance().getAllServiceTypes();
+  }
+
+  public boolean isPrometheusEnabled() {
+
+    return this.getMetrics().stream().anyMatch(metric -> Objects.nonNull(metric.getPrometheus()));
```

Comment:
Since you are checking at some places if metrics is non null do should we check here by doing `.filter(Objects::nonNull)`? 
Although I don't see a reason doing that because each of the config file contains metric. I will leave upto you on whether to remove from above or add the check here.

Reply:
To be honest, I have no idea why I have that `.filter(Objects::nonNull)` on line 77.  I can't think of a time where `this.getMetrics()` would ever contain a null - even if there's no metrics in a config file, `metrics` would initialize as an empty list.

I'm going to remove the unnecessary `.filter(Objects::nonNull)` check